tutorial.md

```diff
@@ -0,0 +1,156 @@
+1. To see current tags, select "View" -> "Cell toolbar" -> "Tags".
+2. Add the tag `noauto` to the cell with the print statement
+3. Add the tag `testing` to the cell with the assert statement
+
+### Save work
+
+Finally, save the notebook.
+
+## Add the example to the table of contents
+
+For building and testing to 'see' a notebook it needs to be in the Jupyterbook table of contents.
+The following steps add the new notebook to the table of contents.
+
+1. Open `idaes_examples/nb/_toc.yml` in an editor.
+2. Find `- file: flowsheets/index` under `chapters:`.
+3. In the list of sections for that chapter, add `- file: flowsheets/my_notebook_doc` (indent the same as the other sections).
+4. Save the modified file.
+
+Note: Although the notebook was called `my_notebook_src`, the filename in the table of contents entry is `my_notebook_doc`.
+
+
+## Test the example
+
+Example notebooks are tested with `pytest`, using a special configuration that is stored in the root directory of the repository (in the file "pyproject.toml").
+
+Change to the top directory for the repository, then test your new example notebook with: 
+```
+pytest -k my_notebook
+```
+
+Why `-k`? Although you can test *all* notebooks with the simple command `pytest`, since notebooks are found and tested with an extra level of indirection you can't simply name the notebook directly. Instead, testing a single notebook requires using a "-k" expression that will match the notebook.
+
+## Run 'unit' tests for all examples
+
+To do quick sanity checks and other tests of the example notebooks, there are standard Python test files under `idaes_examples/nb`. To run those *instead* of the example notebooks, simply run pytest from the `idaes_examples` sub-directory:
+```
+cd idaes_examples
+pytest
+```
+
+Note: This works because there is a "pytest.ini" in that directory that is used instead of the configuration in the root of the repository. That configuration does not set up the special ('nbmake') configuration for running Jupyter notebooks under pytest.
+
+## Rebuild the documentation
+
+To build (or rebuild) the documentation:
+```
+idaesx build
+```
+
+Note #1: This rebuilds *all* the docs, which also involves executing them. This could take many minutes.
+Note #2: From the output you can see that "Jupyterbook" is really doing most of the work here.
+Note #3: In theory rebuilding the docs after the first time can re-use all the previously built versions and save a lot of time. Getting Jupyterbook caching working has been harder than anticipated.
+
+When the documentation finally finished building, the HTML output will be in `idaes_examples\nb\_build\html`. To view, open the "index.html" file with a browser.
+
+## Format the notebooks with Black
+
+To make sure the notebook code is formatted according to IDAES convention, you can run [Black](https://black.readthedocs.io/en/stable/) on all the notebooks:
+
+```
+idaesx black
```

Comment:
I don't see the difference? Are you trying to future-proof it?

Reply:
More for clarity than for future-proofing, i.e. making the command about the what (formatting) than about the how (the specific formatter named Black). In my mind this would be consistent with why we have `idaesx build` instead of `idaesx jupyterbook` or `idaesx gui` instead of `idaesx tkinter`, although I'll admit that it's not exactly a fair comparison since Black has more cachet than the other tools.

That said this is clearly a bikeshed-y comment so feel free to disregard.